components/brave_rewards/browser/rewards_service_factory.cc

```diff
@@ -74,7 +74,7 @@ RewardsServiceFactory::RewardsServiceFactory()
   DependsOn(extensions::EventRouterFactory::GetInstance());
 #endif
 
-#if BUILDFLAG(BRAVE_REWARDS_ENABLED)
+#if BUILDFLAG(BRAVE_REWARDS_ENABLED) && !defined(OS_ANDROID)
```

Comment:
why this change?

Reply:
I don't remember what it does, probably not exist on Android at all

Reply:
I'm quite sure it does exist on android, but this is all wrong anyway and there is a major problem here because we should not be initializing rewards service for private profiles cc @diracdeltas 
RewardsServiceFactory::GetBrowserContextToUse is wrong and needs to be fixed 
```if (context->IsOffTheRecord())
    return chrome::GetBrowserContextOwnInstanceInIncognito(context);

  // use original profile for session profiles
  return chrome::GetBrowserContextRedirectedInIncognito(context);```
should be
```
  // use original profile for session profiles
  return chrome::GetBrowserContextRedirectedInIncognito(context);
```

Reply:
the comment about session profiles doesn't even make sense there and that's not what the method does

Reply:
I'm not sure how that got in there in my commit, but it was definitely an accident

Reply:
in any case we shouldn't be setting any prefs for incognito profiles because we shouldn't be using them at all

Reply:
I'll create a PR for this fix because it apparently originated from my change a long time ago

Reply:
@bridiver should we wait for some PR? Do you want to revert this change?

Reply:
Not sure what PR we need here. I added this define just to make it compilable on Android. Removed from PR.

Reply:
FYI, I created https://github.com/brave/brave-core/pull/3597 to fix RewardsServiceFactory::GetBrowserContextToUse.